homeassistant/components/freebox/base_class.py

```diff
@@ -0,0 +1,197 @@
+"""Support for Freebox base features."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, VALUE_NOT_SET
+from .router import FreeboxRouter
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FreeboxHomeBaseClass(Entity):
+    """Representation of a Freebox base entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant | None,
+        router: FreeboxRouter,
+        node: dict[str, Any],
+        sub_node=None,
```

Comment:
```suggestion
        sub_node: dict[str, Any] | None=None,
```
But this also seems a param for a future use, not used here.

Reply:
Yes preparing code for other devices